nodeup/pkg/model/sysctls.go

```diff
@@ -195,5 +197,38 @@ func (b *SysctlBuilder) Build(c *fi.NodeupModelBuilderContext) error {
 		OnChangeExecute: [][]string{{"sysctl", "--system"}},
 	})
 
+	// Running AWS VPC CNI on Ubuntu 22.04 Requires Setting MACAddressPolicy to `none` (ref: https://github.com/aws/amazon-vpc-cni-k8s/issues/2103 & https://github.com/kubernetes/kops/issues/16255)
+	klog.V(2).Infof("Checking if we need to disable MACAddressPolicy on Ubuntu 22.04")
+	if b.NodeupConfig.Networking.AmazonVPC != nil && b.Distribution == distributions.DistributionUbuntu2204 {
+		klog.V(2).Infof("Detected OS %v; disabling MACAddressPolicy", b.Distribution)
+		// Read the contents of the file /usr/lib/systemd/network/99-default.link into a string
+		origFileName := "/usr/lib/systemd/network/99-default.link"
+		outputFileName := "/etc/systemd/network/99-default.link"
+		input, err := os.ReadFile("/usr/lib/systemd/network/99-default.link")
+		if err != nil {
+			klog.Errorf("Failed to read file %s: %v", origFileName, err)
+			return err
+		}
+
+		// Copy all the relevant entries and replace the one that contains MACAddressPolicy= with MACAddressPolicy=none
+		lines := strings.Split(string(input), "\n")
+		finalOutput := make([]string, 0, len(lines))
+		for i, line := range lines {
+			// We don't want to copy comments or empty lines
+			if strings.HasPrefix(line, "#") || len(line) == 0 {
+				continue
+			} else if strings.Contains(line, "MACAddressPolicy=") {
+				lines[i] = "MACAddressPolicy=none"
+			}
+			finalOutput = append(finalOutput, lines[i])
+		}
+		klog.V(2).Infof("Adding Task to write file %s", outputFileName)
```

Comment:
I think I would prefer a flat file instead, to not be surprised any changes that will be done by Ubuntu.
There is enough logging around tasks. Now that we know it works, we can skip most additional logs.
```suggestion
	lines := []string{
		"[Match]",
		"OriginalName=*",
		"[Link]",
		"NamePolicy=keep kernel database onboard slot path",
		"AlternativeNamesPolicy=database onboard slot path",
		"MACAddressPolicy=none",
	}
	contents := strings.Join(lines, "\n")
```